mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -2269,7 +2274,7 @@ private void resubmitWorker(JobWorker oldWorker) throws Exception {
         @Override
         public int scaleStage(MantisStageMetadataImpl stageMetaData, int numWorkers, String reason) {
             LOGGER.info("Scaling stage {} to {} workers", stageMetaData.getStageNum(), numWorkers);
-            final int oldNumWorkers = stageMetaData.getNumWorkers();
+            final int oldNumWorkers = stageMetaData.getAllWorkers().size();
```

Comment:
Is this safe? There is a valid case where if scaleStage is called multiple times quickly, the worker list is getting updated. So worker list size cannot be used to determine the difference.

Reply:
There is a way to avoid changing this line. Instead of scaling to stageMeta.getNumWorkers, go through the worker list, find the highest worker index from that list, and scale to that.

Reply:
scaleStage being multiple times quickly is not a concern because all the interactions with WorkerManager::scaleStage happen from within the JobActor. Furthermore, Akka guarantees that messages will be handled strictly in serial order. 

I'm afraid I have to disagree with the second idea here. Imagine the following scenario:
1. scaleDown called for the stage from 6 workers to 3 workers.
2. Let's say worker six got removed because workers are removed from 6 to 3 in that order.
3. Now, the machine goes down.
4. We will come back up and realize that three is not equal to 5 (number of active workers).
5. We will scale back to 5 instead of 3, which was the original intention. We are also not scaling back to the original state, which had six workers. 

Because of this, I prefer the current logic that realizes the intent and makes that a reality.